src/sync-pending-events.ts

```diff
@@ -99,29 +119,35 @@ export class EventTracker {
     this.contract = web3.getContract(dataJson, this.address, monitorAccount);
     this.dbUrl = mongodbUrl;
     this.dbName = networkId + this.address;
+    this.source = source;
   }
 
   private async syncPastEvents(
     handlers: (n: string, v: any, hash: string) => Promise<void>,
     db: EventDBHelper
   ) {
-    let lastblock = await db.getLastMonitorBlock();
-    console.log("sync from ", lastblock);
+    // let lastblock = await db.getLastMonitorBlock();
+    let LastCheckedBlock = await db.getLastCheckedBlockNumber();
+    let latestBlockNumber = await getLatestBlockNumber(this.source);
+    console.log("sync from ", LastCheckedBlock);
     try {
-    let pastEvents = await this.contract.getPastEventsFrom(lastblock + 1);
-    console.log("sync from ", lastblock, "done");
-    for (let r of pastEvents) {
-      console.log(
-        "========================= Get L1 Event: %s ========================",
-        r.event
-      );
-      console.log("blockNumber:", r.blockNumber);
-      console.log("blockHash:", r.blockHash);
-      console.log("transactionHash:", r.transactionHash);
-      let e = buildEventValue(this.l1Events, r);
-      await handlers(r.event, e, r.transactionHash);
-      await db.updateLastMonitorBlock(r, e);
+    let pastEvents = await this.contract.getPastEventsFromSteped(LastCheckedBlock, latestBlockNumber, 2000);
+    console.log("sync from ", LastCheckedBlock, "done");
+    for(let group of pastEvents){
+      for (let r of group) {
+        console.log(
+          "========================= Get L1 Event: %s ========================",
+          r.event
+        );
+        console.log("blockNumber:", r.blockNumber);
+        console.log("blockHash:", r.blockHash);
+        console.log("transactionHash:", r.transactionHash);
+        let e = buildEventValue(this.l1Events, r);
+        await handlers(r.event, e, r.transactionHash);
+        await db.updateLastMonitorBlock(r, e);
+      }
     }
+    await db.updateCheckedBlock(latestBlockNumber);
```

Comment:
Here the code have logic change:
1. Originally, the LastCheckedBlock is consistent with the latest event's block number recorded.
2. Now, the LastCheckedBlock will always be the newest block number and not consistent with the latest event's block number.

We need keep original logic after we receive event.
Thus, I suggest we need keep original lastBlockNumber and add a new data like 'ZeroEventCheckedBlockNumber'
Then, the logic need change to be：
if we have "LastBlockNumber != 0 ", we need use the original logic (only update lastBlockNumber when we receive event, so it will consistent with latest record events' block number.)
If "LastBlockNumber == 0 ", it means we never receive any events so we start/continue finding from 'ZeroEventCheckedBlockNumber', and update 'ZeroEventCheckedBlockNumber'.